File: solutions/csharp/sieve/1/Sieve.cs
public static class Sieve
{
    public static int[] Primes(int limit)
    {
        List<int> primes = [];
        bool[] primeStates = Enumerable.Repeat(true, limit + 1).ToArray();
        primeStates[0] = primeStates[1] = false;

        for (int i = 2; i * i <= limit; i++)
            if (primeStates[i])
                for (int j = i * i; j <= limit; j += i)
                    primeStates[j] = false;

        for (int i = 2; i <= limit; i++)
            if (primeStates[i])
                primes.Add(i);
        
        return primes.ToArray();
    }
}

File: solutions/java/sieve/1/src/main/java/Sieve.java
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;

public final class Sieve {
    private int maxPrime;
    
    public Sieve(int maxPrime) {
        this.maxPrime = maxPrime;
    }

    public List<Integer> getPrimes() {
        List<Integer> primes = new ArrayList<>();
        boolean[] primeStates = getInitialPrimeStates();

        for (int i = 2; i * i <= maxPrime; i++)
            if (primeStates[i])
                for (int j = i * i; j <= maxPrime; j += i)
                    primeStates[j] = false;
        
        for (int i = 2; i <= maxPrime; i++)
            if (primeStates[i]) 
                primes.add(i);

        return primes;
    }

    private boolean[] getInitialPrimeStates() {
        boolean[] primeStates = new boolean[maxPrime + 1];
        
        Arrays.fill(primeStates, true);
        primeStates[0] = false;
        primeStates[1] = false;
        
        return primeStates;
    }
}
